### PR TITLE
Add piece to onSquareClicked and square to onPieceClicked

### PR DESCRIPTION
**Pull request by KeeghanM**

### CODE REVIEW

File: src/chessboard/components/Square.tsx [hunk 1]
```diff
@@ -150,7 +150,10 @@ export function Square({
       }}
       onDragEnter={() => onDragOverSquare(square)}
       onClick={() => {
-        onSquareClick(square);
+        const piece = Array.isArray(children)
+          ? (children.find((c) => c?.props?.piece)?.props?.piece as Piece)
+          : undefined;
```
Comment (Manukyanq) [inline src/chessboard/components/Square.tsx:155]: get piece by square is much easier 
```suggestion
        const piece = currentPosition[square]
```

**Comment by Manukyanq** [comment]
don't forget to update the description of functions `onSquareClick ` and `onPieceClick` in `README.mdx`

File: src/chessboard/components/Square.tsx (hunk 1 above)
Comment (KeeghanM) [reply src/chessboard/components/Square.tsx]: wow. Yeah didn't realise that was there! Much simpler

**Comment by Clariity** [comment]
FIxes #127

**Review by Clariity — APPROVED** [review]
Thank you

**Pull request merged**